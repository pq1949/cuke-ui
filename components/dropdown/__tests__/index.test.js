import React from "react";
import assert from "power-assert";
import { render, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Dropdown from "../index";
import Popover from "../../popover";

const overlay = (
  <ul>
    <li> 红烧牛肉 </li>
    <li> 麻辣干锅 </li>
    <li> 仔姜肉丝 </li>
  </ul>
);

describe("<Dropdown/>", () => {
  it("should render a <Dropdown/> components", () => {
    const wrapper = render(
      <div>
        <Dropdown overlay={overlay}>
          <a href="#">今日菜单(有动画)</a>
        </Dropdown>
        <Dropdown overlay={overlay} animate={false}>
          <a href="#">今日菜单(无动画)</a>
        </Dropdown>
      </div>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should find cuke-dropdown classnames", () => {
    const wrapper = render(
      <div>
        <Dropdown overlay={overlay}>
          <a href="#">今日菜单(有动画)</a>
        </Dropdown>
        <Dropdown overlay={overlay} animate={false}>
          <a href="#">今日菜单(无动画)</a>
        </Dropdown>
      </div>
    );
    assert(wrapper.find(".cuke-dropdown").length >= 1);
  });

  it("should find <Popover/>", () => {
    const wrapper = shallow(
      <Dropdown overlay={overlay}>
        <a href="#">今日菜单(有动画)</a>
      </Dropdown>
    );
    assert(wrapper.find(Popover).length === 1);
  });
});
