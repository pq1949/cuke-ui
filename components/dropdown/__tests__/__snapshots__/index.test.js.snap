// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown/> should render a <Dropdown/> components 1`] = `
<div>
  <div
    class="cuke-popover cuke-dropdown"
  >
    <div
      class="cuke-tooltip"
    >
      <div
        class="cuke-tooltip-wrapper cuke-tooltip-position-bottom cuke-tooltip-light cuke-popover-wrapper cuke-dropdown-wrapper cuke-tooltip-hide cuke-ui-no-animate"
        style="left:0;top:0"
      >
        <div
          class="cuke-popover-content"
        >
          <ul>
            <li>
               红烧牛肉 
            </li>
            <li>
               麻辣干锅 
            </li>
            <li>
               仔姜肉丝 
            </li>
          </ul>
        </div>
      </div>
      <span
        class="cuke-tooltip-trigger-wrapper"
      >
        <a
          href="#"
        >
          今日菜单(有动画)
        </a>
      </span>
    </div>
  </div>
  <div
    class="cuke-popover cuke-dropdown"
  >
    <div
      class="cuke-tooltip"
    >
      <div
        class="cuke-tooltip-wrapper cuke-tooltip-position-bottom cuke-tooltip-light cuke-popover-wrapper cuke-dropdown-wrapper cuke-tooltip-hide cuke-ui-no-animate"
        style="left:0;top:0"
      >
        <div
          class="cuke-popover-content"
        >
          <ul>
            <li>
               红烧牛肉 
            </li>
            <li>
               麻辣干锅 
            </li>
            <li>
               仔姜肉丝 
            </li>
          </ul>
        </div>
      </div>
      <span
        class="cuke-tooltip-trigger-wrapper"
      >
        <a
          href="#"
        >
          今日菜单(无动画)
        </a>
      </span>
    </div>
  </div>
</div>
`;
